Library/Homebrew/utils/ruby.sh

```diff
@@ -45,15 +57,34 @@ EOS
       then
         HOMEBREW_RUBY_PATH="/System/Library/Frameworks/Ruby.framework/Versions/Current/usr/bin/ruby"
       else
-        HOMEBREW_RUBY_PATH="$(type -P ruby)"
+        HOMEBREW_RUBY_PATH=$(type -P ruby)
+        if [[ $(test-ruby $HOMEBREW_RUBY_PATH) != "true" ]]
+        then
+          HOMEBREW_RUBY_PATH=$(PATH="$HOMEBREW_PATH" type -P ruby)
+          if [[ $(test-ruby "$HOMEBREW_RUBY_PATH") != "true" ]]
+          then
+            if [[ $old_ruby_usable != true ]]
+            then
+              odie <<-EOS
+Failed to find usable Ruby $required_ruby_version!
```

Comment:
Would be nice to move this new logic below so the messaging can be shared rather than repeated.